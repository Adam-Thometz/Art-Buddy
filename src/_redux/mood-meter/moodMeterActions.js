import { createAction } from "@reduxjs/toolkit";

export const setValence = createAction('setValence');
export const setEnergy = createAction('setEnergy');
export const clearOptions = createAction('clearOptions');