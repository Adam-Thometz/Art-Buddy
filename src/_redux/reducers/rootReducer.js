// import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';

import mainSettingsReducer from './mainSettingsReducer';
import wordToMusicDecoderReducer from './wordToMusicDecoderReducer';
import sequenceMakerReducer from './sequenceMakerReducer';
import scoreKeeperReducer from './scoreKeeperReducer';
import instrumentIdReducer from './instrumentIdReducer';

const rootReducer = configureStore({
  reducer: {
    mainSettings: mainSettingsReducer,
    wordToMusic: wordToMusicDecoderReducer,
    sequenceMaker: sequenceMakerReducer,
    scoreKeeper: scoreKeeperReducer,
    instrumentId: instrumentIdReducer
  }
});

export default rootReducer;