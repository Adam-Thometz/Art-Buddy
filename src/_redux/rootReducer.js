import { configureStore } from '@reduxjs/toolkit';

import mainSettingsReducer from './settings/mainSettingsReducer';
import musicDecoderReducer from './music-decoder/musicDecoderReducer';
import jumpIntoRhythmReducer from './jump-into-rhythm/jumpIntoRhythmReducer';
import sequenceMakerReducer from './sequence-maker/sequenceMakerReducer';
import idListeningSkillsTestReducer from './instrument-id/listening-skills/idListeningSkillsReducer';
import idSongMakerReducer from './instrument-id/song-maker/idSongMakerReducer';
import freePaintReducer from './free-paint/freePaintReducer';
import colorTheoryReducer from './color-theory/colorTheoryReducer';
import scoreKeeperReducer from './score-keeper/scoreKeeperReducer';

const rootReducer = configureStore({
  reducer: {
    mainSettings: mainSettingsReducer,
    musicDecoder: musicDecoderReducer,
    sequenceMaker: sequenceMakerReducer,
    listeningSkillsTest: idListeningSkillsTestReducer,
    songMaker: idSongMakerReducer,
    scoreKeeper: scoreKeeperReducer,
    jumpIntoRhythm: jumpIntoRhythmReducer,
    freePaint: freePaintReducer,
    colorTheory: colorTheoryReducer
  }
});

export default rootReducer;