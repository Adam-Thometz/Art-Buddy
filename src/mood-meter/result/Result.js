import React from 'react';

import { useSelector } from 'react-redux';

import './Result.css';

import Face from 'mood-meter/valence/face/Face';
import Battery from 'mood-meter/energy/battery/Battery';

import faces from '_data/mood-meter/faces';
import batteries from '_data/mood-meter/batteries';
import moods from '_data/mood-meter/moods';
import Button from '_components/button/Button';

const Result = () => {
  const { valence, energy } = useSelector(state => state.moodMeter);

  const face = faces.find(face => face.id === valence);
  const valenceIcon = <Face key={face.id} src={face.src} id={face.id} isSelected={true} />;
  
  const battery = batteries.find(battery => battery.id === energy);
  const energyIcon = <Battery key={battery.id} width={battery.width} id={battery.id} filter={battery.filter} />;
  
  const mood = moods[valence][energy];
  
  return (
    <div className='Result'>
      <div className='Result-icons'>
        {valenceIcon}
        {energyIcon}
      </div>
      <p className='Result-text'>{mood}</p>
      <Button colorId={2}>Listen To Song</Button>
    </div>
  );
};

export default Result;