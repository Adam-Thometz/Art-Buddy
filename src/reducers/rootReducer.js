import { combineReducers } from 'redux';
import musicColorsReducer from './musicColorsReducer';
import sequenceMakerReducer from './sequenceMakerReducer';

const rootReducer = combineReducers({
  musicColors: musicColorsReducer,
  sequenceMaker: sequenceMakerReducer
});

export default rootReducer;