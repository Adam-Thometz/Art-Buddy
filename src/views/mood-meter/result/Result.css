.Result {
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.Result-icons {
  display: flex;
  justify-content: space-evenly;
  width: 40%;
  margin: 0 auto;
}

.Result-text {
  font-size: 5rem;
  color: #17255a;
}

.Result .Button {
  width: 30%;
  height: 10vh;
  margin: 0 auto;
}
