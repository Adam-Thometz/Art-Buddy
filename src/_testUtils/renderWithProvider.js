import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;