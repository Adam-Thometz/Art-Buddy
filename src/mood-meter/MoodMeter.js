import React, { useContext, useEffect } from 'react';
import useVisited from '_hooks/visited/useVisited';
import { PopupContext } from '_context/PopupContext';

import { useDispatch, useSelector } from 'react-redux';
import { changeCurrGame } from '_redux/_general/generalActions';

import WindowNavbar from '_components/window-nav/WindowNavbar';
import Instructions from '_components/instructions/Instructions';
import Valence from './valence/Valence';
import Energy from './energy/Energy';

import { moodMeter } from '_data/_activities/activityList';
import Result from './result/Result';

const MoodMeter = () => {
  const [hasVisited] = useVisited(moodMeter.lsKey);
  const { valence, energy } = useSelector(state => state.moodMeter);
  const { currGame } = useSelector(state => state.general);
  const { setCurrPopup } = useContext(PopupContext)
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(changeCurrGame(moodMeter));
    return () => {
      dispatch(changeCurrGame({}));
    };
  }, [dispatch]);

  useEffect(() => {
    if (valence !== null && energy !== null) {
      setCurrPopup({
        title: 'YOUR MOOD IS...',
        popup: <Result />
      });
    };
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [valence, energy]);

  return (<>
    <WindowNavbar page={currGame.name} />
    {!hasVisited ? <Instructions /> : <>
      <Valence />
      <Energy />
    </>}
  </>);
};

export default MoodMeter;