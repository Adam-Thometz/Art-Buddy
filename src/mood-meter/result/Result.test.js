import React from "react";
import { default as store } from "_redux/rootReducer";
import { setEnergy, setValence } from "_redux/mood-meter/moodMeterActions";

import renderWithProvider from "_testUtils/renderWithProvider";
import Result from "./Result";

describe('Result component', () => {
  store.dispatch(setValence(2));
  store.dispatch(setEnergy(2));
  it('renders without crashing', () => {
    renderWithProvider(<Result />);
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProvider(<Result />);
    expect(asFragment()).toMatchSnapshot();
  });
});