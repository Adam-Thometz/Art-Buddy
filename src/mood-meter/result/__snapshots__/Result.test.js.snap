// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Result"
  >
    <div
      class="Result-icons"
    >
      <div
        class="Face selected-face"
        data-testid="face2"
        id="2"
      >
        <img
          alt=""
          id="2"
          src="file"
        />
      </div>
      <div
        class="Battery"
        data-testid="battery2"
        id="2"
      >
        <img
          alt=""
          id="2"
          src="file"
        />
        <img
          alt=""
          data-testid="bar"
          id="2"
          src="file"
          style="width: 89%; filter: brightness(0%) saturate(100%) invert(68%) sepia(56%) saturate(915%) hue-rotate(38deg) brightness(109%) contrast(92%);"
        />
      </div>
    </div>
    <p
      class="Result-text"
    >
      Excited
    </p>
    <button
      class="Button regular"
      style="border-color: #fd4d3a;"
    >
      Listen To Song
    </button>
  </div>
</DocumentFragment>
`;
